cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -569,20 +599,13 @@ func (m *stateMachine) mountBoot() (stateFunc, error) {
 	fsckSystemdOpts := &systemdMountOptions{
 		NeedsFsck: true,
 	}
-	if err := doSystemdMount(dev, boot.InitramfsUbuntuBootDir, fsckSystemdOpts); err != nil {
-		// didn't manage to mount boot, so try to use fallback key for data
-		m.degradedState.LogErrorf("failed to mount ubuntu-boot: %v", err)
-		m.degradedState.partition("ubuntu-boot").MountState = partitionErrMounting
-		return m.unlockDataFallbackKey, nil
-	}
-
-	// verify ubuntu-boot comes from same disk as ubuntu-seed
-	if err := m.verifyMountPoint(boot.InitramfsUbuntuBootDir, "ubuntu-boot"); err != nil {
+	mountErr := doSystemdMount(dev, boot.InitramfsUbuntuBootDir, fsckSystemdOpts)
+	if err := m.setMountState("ubuntu-boot", boot.InitramfsUbuntuBootDir, mountErr); err != nil {
 		return nil, err
 	}
-
-	m.degradedState.partition("ubuntu-boot").MountState = partitionMounted
-	m.degradedState.partition("ubuntu-boot").MountLocation = boot.InitramfsUbuntuBootDir
+	if m.degradedState.partition("ubuntu-boot").MountState == partitionErrMounting {
+		return m.unlockDataFallbackKey, nil
```

Comment:
it would be nice to have a small comment explaining the transition here, i.e.

```go
// if we didn't mount data, then try to unlock data with the fallback key
```